transform/operations_test.go

```diff
@@ -375,21 +400,108 @@ func TestTimeParse(t *testing.T) {
 	runOpTests(t, func() transformOperation { return &timeParse{} }, tests)
 }
 
+func TestCurrentTime(t *testing.T) {
+	tests := []opTests{
+		{
+			description: "Simple working case",
+			args:        map[string]string{"format": "RFC3339"},
+			want:        time.Now().Format(time.RFC3339),
+		},
+		{
+			description: "Missing arg",
+			args:        map[string]string{},
+			wantInitErr: true,
+		},
+		{
+			description: "Too many args",
+			args:        map[string]string{"format": "RFC3339", "cookies": "failure"},
+			wantInitErr: true,
+		},
+		{
+			description: "Format not predefined",
+			args:		 map[string]string{"format": "Mon Jan 2 15:04:05 MST 2006"},
+			want:		time.Now().Format("Mon Jan 2 15:04:05 MST 2006")	,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			op, err := runOpTestInit(func() transformOperation { return &currentTime{} }, test)
+			if err != nil && test.wantInitErr {
+				return
+			}
+			if err != nil && !test.wantInitErr {
+				t.Fatal(err)
+			}
+			got, err := runOpTestTransform(op, test)
+
+			if err != nil {
+				t.Fatal(err)
+				return
+			}
+			now := time.Now()
+			result, ok := got.(string)
+			if !ok {
+				t.Fatal("function must return string")
+			}
+
+			var gotParse time.Time
+
+			if test.args["format"] == "RFC3339" {
```

Comment:
It would be better to put the expected time in the test case rather than calculating it down here

Reply:
This still needs resolved.

Something like

```golang
wantTime, err := time.Parse(test.args["format"], wantResult)
gotTime, err := time.Parse(test.args["format"], gotResult)
```

then compare the two

Reply:
Or change your `test.want` to be a time object instead of a string and then you only have to do the time.Parse once